lib/players.rb

```diff
@@ -9,20 +10,22 @@ def initialize(ui)
   end
 
   def codemaker(input)
-    if input == "computer"
+    if input == "c"
       Computer.new("computer-codemaker", 4)
-    elsif input == "human player"
+    elsif input == "h"
       HumanPlayer.new(@ui.ask_human_player_name, @ui, 4)
     else
       raise("I didn't understand :(")
     end
   end
 
-  def codebreaker(input)
-    if input == "computer"
+  def codebreaker(input, board)
+    if input == "c"
       Computer.new("computer-codebreaker", 4)
-    elsif input == "human player"
+    elsif input == "h"
       HumanPlayer.new(@ui.ask_human_player_name, @ui, 4)
+    elsif input == "s"
```

Comment:
inconsistent naming here - "smart computer" vs. "unbeatable computer". Pick one and stick to it